app/controllers/fhir/r4/api_controller.rb

```diff
@@ -457,6 +459,27 @@ def current_client_application
     OauthApplication.find_by(id: doorkeeper_token.application_id) if doorkeeper_token.application_id.present?
   end
 
+  # Determine if the patient's jurisdiction is valid for the requesting application
+  def jurisdiction_valid_for_client?(patient)
+    allowed_jurisdiction_ids = current_client_application.jurisdiction&.subtree&.pluck(:id) ||
+                               current_resource_owner&.jurisdiction&.subtree&.pluck(:id)
+
+    if allowed_jurisdiction_ids.nil?
+      patient.errors.add(:jurisdiction_id, 'Client application does not have a jurisdiction')
```

Comment:
Not necessarily the client application, this would mean for the user workflow that the user doesn't have a jurisdiction (which shouldn't be possible, but still).

Reply:
For this I was thinking the wording still makes sense though. From the client application's perspective, its jurisdiction _is_ just the user-that-created-it's jurisdiction, isn't it? It's easy to change though so I'll just rework this slightly to allow for different messaging per auth workflow.

Reply:
So for a user using the user workflow, their user account is not directly related to the client application in any way. That user account did not create the OAuth app. For the m2m workflow, a shadow user is directly specified and a jurisdiction ID explicitly said. Does that make sense?

Reply:
Right, yeah my bad I'm with you now.